code/FulpstationCode/sec_borg_files/fulp_sec_borg.dm

```diff
@@ -0,0 +1,477 @@
+//SEC INTEGRATED E-BOLA (lol) LAUNCHER ITEM AND UPGRADE - BEGINS
+//*******************************************
+
+/obj/item/gun/energy/e_gun/e_bola/cyborg
+	name = "\improper Integrated E-BOLA Launcher"
+	desc = "An integrated e-bola launcher that draws from a cyborg's power cell."
+	icon_state = "dragnet"
+	can_charge = FALSE
+	use_cyborg_cell = TRUE
+	charge_delay = 8
+	ammo_type = list(/obj/item/ammo_casing/energy/bola)
+
+/obj/item/borg/upgrade/e_bola
+	name = "cyborg energy bola launcher"
+	desc = "A module that permits firing energy bolas."
+	icon = 'icons/obj/guns/energy.dmi'
+	icon_state = "dragnet"
+	require_module = TRUE
+	module_type = list(/obj/item/robot_module/security)
+
+/obj/item/borg/upgrade/e_bola/action(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if(.)
+		var/obj/item/gun/energy/e_gun/e_bola/cyborg/E = locate() in R.module.modules
+		if(E)
+			to_chat(user, "<span class='warning'>This unit already has a [E] installed!</span>")
+			return FALSE
+
+		E = new(R.module)
+		R.module.basic_modules += E
+		R.module.add_module(E, FALSE, TRUE)
+
+/obj/item/borg/upgrade/e_bola/deactivate(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if (.)
+		var/obj/item/gun/energy/e_gun/e_bola/cyborg/E = locate() in R.module.modules
+		if (E)
+			R.module.remove_module(E, TRUE)
+
+/obj/item/ammo_casing/energy/bola
+	projectile_type = /obj/projectile/energy/trap/cyborg
+	select_name = "bola"
+	e_cost = 400
+	harmful = FALSE
+
+//*******************************************
+//SEC INTEGRATED E-BOLA (lol) LAUNCHER ITEM AND UPGRADE - ENDS
+//*******************************************
+
+//*******************************************
+//SEC INTEGRATED ENERGY GUN ITEM AND UPGRADE - BEGINS
+//*******************************************
+
+/obj/item/gun/energy/e_gun/cyborg
+	name = "\improper Integrated Energy Gun"
+	desc = "An integrated energy gun that draws from a cyborg's power cell."
+	can_charge = FALSE
+	use_cyborg_cell = TRUE
+
+/obj/item/borg/upgrade/e_gun
+	name = "cyborg energy gun"
+	desc = "A module that equips the unit with an energy gun."
+	icon = 'icons/obj/guns/energy.dmi'
+	icon_state = "energy"
+	require_module = TRUE
+	module_type = list(/obj/item/robot_module/security)
+
+/obj/item/borg/upgrade/e_gun/action(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if(.)
+		var/obj/item/gun/energy/e_gun/cyborg/E = locate() in R.module.modules
+		if(E)
+			to_chat(user, "<span class='warning'>This unit already has a [E] installed!</span>")
+			return FALSE
+
+		E = new(R.module)
+		R.module.basic_modules += E
+		R.module.add_module(E, FALSE, TRUE)
+
+/obj/item/borg/upgrade/e_gun/deactivate(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if (.)
+		var/obj/item/gun/energy/e_gun/cyborg/E = locate() in R.module.modules
+		if (E)
+			R.module.remove_module(E, TRUE)
+
+/obj/item/gun/energy/e_gun/cyborg/can_shoot()
+	var/obj/item/ammo_casing/energy/shot = ammo_type[select]
+	if(GLOB.security_level < SEC_LEVEL_RED && shot.harmful)

```

Comment:
if they're emagged, they shouldn't care about whether or not the alert level is less than red

Reply:
See below.